app/components/learnergroup-members.js

```diff
@@ -122,38 +134,106 @@ export default Ember.Component.extend({
-            topLevelGroup.removeUserFromGroupAndAllDescendants(user).then(groups=>{
-              toSave.pushObjects(groups);
-              if(groupId === -1){
-                //we're moving this user out of the group into the cohort
-                //so just save
-                Ember.RSVP.all(toSave.uniq().invoke('save')).then(()=>{
-                  this.set('saving', false);
-                });
-              } else {
-                this.get('store').find('learnerGroup', groupId).then( learnerGroup => {
-                  learnerGroup.addUserToGroupAndAllParents(user).then(groups =>{
-                    toSave.pushObjects(groups);
-                    Ember.RSVP.all(toSave.uniq().invoke('save')).then(()=>{
-                      this.set('saving', false);
-                    });
+
+  sendPutRequests(groupIdString, userId) {
+    this.set('saving', true);
+    let groupId = parseInt(groupIdString);
+    let toSave = [];
+    let component = this;
+
+    this.get('store').find('user', userId).then(
+      (user) => {
+        toSave.pushObject(user);
+        component.get('topLevelGroup').then((topLevelGroup) => {
+          topLevelGroup.removeUserFromGroupAndAllDescendants(user).then((groups) => {
+            toSave.pushObjects(groups);
+            if (groupId === -1) {
+              // we're moving this user out of the group into the cohort
+              // so just save
+              all(toSave.uniq().invoke('save')).finally(() => {
+                component.set('saving', false);
+              });
+            } else {
+              component.get('store').find('learnerGroup', groupId).then(learnerGroup => {
+                learnerGroup.addUserToGroupAndAllParents(user).then(groups =>{
+                  toSave.pushObjects(groups);
+                  all(toSave.uniq().invoke('save')).finally(() => {
+                    component.set('saving', false);
                   });
                 });
-              }
-              
-            });
+              });
+            }
           });
-        }
-      );
+        });
+      }
+    );
+  },
+
+  actions: {
+    changeLearnerGroup(groupIdString, userId){
+      this.sendPutRequests(groupIdString, userId);
+    },
+
+    addStudent(studentId) {
+      this.get('toBulkSave').pushObject(studentId);
+    },
+
+    removeStudent(studentId) {
+      this.get('toBulkSave').removeObject(studentId);
+    },
+
+    continueButton() {
+      if (isEmpty(this.get('toBulkSave'))) {
+        this.set('errorMessage', 'You must checkmark at least one.');
+      } else {
+        this.setProperties({ errorMessage: null, editorModeOn: true });
+      }
+    },
+
+    changeSelection(newValue) {
+      newValue = newValue.get('value') === 'null' ? null : newValue.get('value');
+      this.send('changeValue', newValue);
+    },
+
+    bulkSave(value) {
+      this.get('toBulkSave').forEach((studentId) => {
+        this.sendPutRequests(value, studentId);
```

Comment:
I think there is a lot of potential for a race condition here.  As each student saves it also sends a put request with all the learner groups and all of their students.  Depending on the order it will be possible for a student to be removed once and then added back again when the next student saves.  Its possible that ember-date will save this from happening, but I'm not sure if that is true.